torchao/quantization/linear_activation_weight_observed_tensor.py

```diff
@@ -147,6 +148,8 @@ def _(func, types, args, kwargs):
         args[0].to(*args[1:], **kwargs)._apply_fn_to_data(torch.clone),
     )
 
+to_linear_activation_weight_observed = LinearActivationWeightObservedTensor.from_float
```

Comment:
one thing I dont like about this pattern is that it makes doc finding/IDE things worse

Reply:
this is to be consistent with other things right now, i.e. we use factory functions to create tensors. for this tensor subclass specifically, it's more of a "util" tensor subclass, maybe we can have different rules, like for "dtype tensor subclass" we use `to_tensor_subclass` to create them, for "util" tensor subclasses we use `TensorSubclass.from_float` etc. but I'm not sure if it's worth it to distinguish between different types here

Reply:
I'm not very familiar with doc finding/IDE, any way to mitigate?

Reply:
From some googling I dont think there are many ways to mitigate. 
>this is to be consistent with other things right now, i.e. we use factory functions to create tensors. 
Like in my mind Object.method is the canonical way to define a factory function in Python. I never really understood why we did this function name aliasing? is it just to make it shorter that feels weird especially since its an internal dev facing api and not end model user facing. 

I dont need to die on this hill but if this is all for naming I am find with changing `from_float` to any other name you would prefer.

Reply:
it's for following things like https://pytorch.org/docs/stable/generated/torch.Tensor.to_sparse.html#torch.Tensor.to_sparse, https://pytorch.org/docs/stable/generated/torch.Tensor.to_sparse_bsc.html#torch.Tensor.to_sparse_bsc, https://pytorch.org/docs/stable/generated/torch.Tensor.to_sparse_coo.html#torch.Tensor.to_sparse_coo I think.

maybe this will help explain: https://docs.google.com/document/d/1vXVQEM8FKKF73vNUsSfG2QrkuD5z3Y3eYyj-qcrdJX4/edit?tab=t.0#bookmark=id.92qvtqebp9wz

I also don't care about this as much, I can change it back